File: domain/src/main/java/com/study/platform/domain/team/model/StudyTeamRepository.java
package com.study.platform.domain.team.model;

import java.util.Optional;
import java.util.UUID;

public interface StudyTeamRepository {

    StudyTeam save(StudyTeam team);
    Optional<StudyTeam> findById(UUID id);
    void delete(StudyTeam team);
    Optional<StudyTeam> findByPostId(UUID postId);
}

File: domain/src/main/kotlin/com/study/platform/domain/team/model/StudyTeamRepository.kt
package com.study.platform.domain.team.model

import java.util.UUID

interface StudyTeamRepository {

    fun save(team: StudyTeam): StudyTeam
    fun findById(id: UUID): StudyTeam?
    fun delete(team: StudyTeam)
    fun findByPostId(postId: UUID): StudyTeam?
}
